src/main/java/sg/ncl/MainController.java

```diff
@@ -1796,10 +1820,37 @@ public String teamAppSubmitFromTeamsPage() {
         return "team_page_application_submitted";
     }
     
-    @RequestMapping("/teams/join_application_submitted/{teamId}")
-    public String teamAppJoinFromTeamsPage(@PathVariable Integer teamId, Model model) {
-        int teamOwnerId = teamManager.getTeamByTeamId(teamId).getTeamOwnerId();
-        model.addAttribute("teamOwner", userManager.getUserById(teamOwnerId));
+    @RequestMapping("/teams/join_application_submitted/{teamName}")
+    public String teamAppJoinFromTeamsPage(@PathVariable String teamName, Model model) throws WebServiceRuntimeException {
+        logger.info("Join application submitted");
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.set("Authorization", AUTHORIZATION_HEADER);
+
+        HttpEntity<String> request = new HttpEntity<>("parameters", headers);
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        ResponseEntity response = restTemplate.exchange(properties.getTeamByName(teamName), HttpMethod.GET, request, String.class);
+
+        String responseBody = response.getBody().toString();
+
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+
+                if (error.getName().equals(ExceptionState.TeamNotFoundException.toString())) {
+                    logger.warn("submitted join team request : team name error");
+                } else {
+                    logger.warn("submitted join team request : some other failure");
+                    // possible sio or adapter connection fail
+                }
+                return "redirect:/teams/join_team";
+            }
+        } catch (IOException e) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)